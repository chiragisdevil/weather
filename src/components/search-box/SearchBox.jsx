/* SearchBox Component takes the following props:
        1) Placeholder - Which will be displayed as the placeholder
        2) setSearchedItem - Generic function that sets the state in the parent component. Can be used to search a city 
        & later on for other fields as well
        3) navigate - This will be the route that we want to route to if clicked. This will be defaulted to "/" if null
*/

import React, {useState} from "react";
import {Link, withRouter} from "react-router-dom";
import { BsSearch } from "react-icons/bs";
import "./search-box.scss";

function SearchBox(props){
    //Local State as we'd like to set the SearchedItem state only when search icon is clicked
    const [item, setItem] = useState();
    const path = (props.navigate?props.navigate:"/");
    return (
        <div>
                <input type="search" className="search-bar" placeholder = { props.placeholder} onChange={(event) => setItem(event.target.value)} />
                <Link to= {path}>
                    <BsSearch className="search-icon" onClick={() => props.setSearchedItem(item)}/>
                </Link>
       </div>
    )
}

export default withRouter(SearchBox);