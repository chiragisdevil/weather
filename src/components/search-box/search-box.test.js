// import { EnzymeAdapter } from "enzyme";
import {shallow, mount, render} from "enzyme";
import SearchBox from "./SearchBox";

describe("SearchBox Component tests", () => {

    function setCity(city){
        return(
            console.log(city)
        )
    }

it("checks if SearchBox container component matches the snapshot", () => {
    
    expect(shallow(<SearchBox 
        placeholder = "London"
        setSearchedItem = {setCity}
        navigate = {"/weather"}
    />)).toMatchSnapshot();
})

it('should call onChange prop', () => {
    const city = "London";
    const path="/";
    const wrapper = shallow(<SearchBox.WrappedComponent setSearchedItem = {setCity} placeholder = {city} navigate = {path}/>);
    const instance = wrapper.instance();
    // const onChange = jest.fn();
    const event = {
      target: { value: "London" }
    };
    wrapper.find('input').simulate('change', event);

    expect(wrapper.find('input').prop('value')).toEqual(
        'London',
      );
  });

});