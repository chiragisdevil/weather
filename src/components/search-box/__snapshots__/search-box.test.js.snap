// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component tests checks if SearchBox container component matches the snapshot 1`] = `ShallowWrapper {}`;
